src/averageAges.js

```diff
@@ -31,13 +46,25 @@ function calculateMenAverageAge(people, century) {
  * Hint: To check if a woman has children you should find someone who mention
  * her as mother.
  *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
+//  * @param {object[]} people
+//  * @param {boolean} withChildren - optional
+//  *
+//  * @return {number}
+//  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWoman = withChildren
+    ? people
+      .filter(person => (
+        people.find(child => child.mother === person.name && person.sex === 'f')
+      ))
+    : people
+      .filter(person => person.sex === 'f');
```

Comment:
You use a filter in both cases. Maybe it's better to do a check `withChildren` condition inside the filter?